src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -695,9 +695,22 @@ internal void ExecutePackageDownload(string name, PackageVersion package, string
                             Name = dep.name
                         };
                     })
-                    .ToList()
-                    .ForEach(x => DownloadAndInstall(x, downloadPath));
+                    .Select(x => Download(x)).ToArray();
+                //wait for all downloads.
+                await Task.WhenAll(downloadTasks);
```

Comment:
Any chance that a download task can throw an exception?

Reply:
thats a good point, it probably can, need to go read up on aggregate exceptions and interaction with `WhenAll`

Reply:
ok I checked this out - the inner method `DownloadPackage` does not throw, and instead this method will return a modified handle that is marked with `Error` - so I think we're good. I am writing a test anyway though to make sure split into download/install here does not cause issues.